### PR TITLE
[SPARK-38961][DOCS][PYTHON][3.3] Enhance to automatically generate the pandas API support list

### PR DESCRIPTION
**Pull request by beobest2**
### What changes were proposed in this pull request?

This PR proposes to automatically generate a list of supported APIs when building a page called "Supported pandas APIs" documentation.

### Why are the changes needed?

Supported pandas APIs list is maintained manually, so it is recommended to generate the list automatically to reduce maintenance costs.

### Does this PR introduce _any_ user-facing change?

Yes, the "Supported pandas APIs" page has changed as below. A list of APIs is created automatically.
Screen Shot 2022-05-30 at 10 51 12 PM

### How was this patch tested?

Manually check the links in the documents & the existing doc build should be passed.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
Quick question. Does https://github.com/apache/spark/blob/branch-3.3/python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst in branch-3.3 have the same issue of broken links? If that's the case, let's backport this together.

If branch-3.3 does not have the issue of broken links, let's backport https://github.com/apache/spark/pull/36729 alone.

File: python/pyspark/pandas/supported_api_gen.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    Generate supported APIs status dictionary.
+
+    Parameters
+    ----------
+    output_rst_file_path : str
+        The path to the document file in RST format.
+
+    Write supported APIs documentation.
+    """
+    if LooseVersion(pd.__version__) < LooseVersion("1.4.0"):
```
Comment (Yikun) [inline python/pyspark/pandas/supported_api_gen.py:100]: If target version is Spark `3.3.0`, we might want to pin version to `1.3.5`.

File: python/pyspark/pandas/supported_api_gen.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+
+* 'Y' in the second column means it's implemented including its whole parameter.
+* 'N' means it's not implemented yet.
+* 'P' means it's partially implemented with the missing of some parameters.
+
+All API in the list below computes the data with distributed execution except the ones that require
+the local execution by design. For example, `DataFrame.to_numpy() <https://spark.apache.org/docs/
+latest/api/python/reference/pyspark.pandas/api/pyspark.pandas.DataFrame.to_numpy.html>`__
+requires to collect the data to the driver side.
+
```
Comment (Yikun) [inline python/pyspark/pandas/supported_api_gen.py:62]: https://github.com/apache/spark/blame/branch-3.3/python/docs/source/user_guide/pandas_on_spark/supported_pandas_api.rst#L39

FYI, some diff between 3.3 and original master.

**Review by Yikun — COMMENTED** [review]
Should we remove the `supported_pandas_api.rst`? Because it would be generate automically.

**Comment by beobest2** [comment]
> #36729 alone.

@HyukjinKwon  Oops, I misunderstood that auto-generated code and documentation edits were included in the backport together. The reason #36729 alone is difficult to apply to `branch-3.3` is that the top part of the document is included in the auto-generated code (supported_api_gen.py) and the existing `supported_pandas_api.rst` has been deleted at https://github.com/apache/spark/commit/5fab58f0429a08d1a303f764664d7e5a3badc415.

**Comment by Yikun** [comment]
I guess @HyukjinKwon only mean fix the typo on static doc? https://github.com/apache/spark/pull/36729#issuecomment-1144473696 I'm not sure can we land all auto-generate script on `3.3`. : )

[1] https://lists.apache.org/thread/m2h6hrg4x9z1xwmg6fz271g86q05vg79

**Comment by beobest2** [comment]
@Yikun That's right! I just re-created the PR (https://github.com/apache/spark/pull/36749) by correcting typos in the static documentation. 

In the case of automatic document generation, it seems that it will take some time for the broken link issue to be resolved. 
Thanks for always helping with reviews :)

**Pull request closed without merging**